### ISSUE TITLE
Set authentication database

### ISSUE THREAD

**Issue by FroznMine** [sboesebeck/morphium#20]
I need the possibility to set a database for authentication which differs from the one containing all the data Morphium should interact with. In MongoDB's Java driver this is possible by specifying a authentication database in the connection string. As far as I know Morphium uses the same database for authentication and data. Maybe I'm wrong?

**Comment by sboesebeck** [comment]
Hi...

I'm not sure, I get you right: you can set different credentials for different databases. This is the same in Morphium and Mongodb's Driver. I don't think you can authenticate against db1 and use db2 or something (unless db2 does not have authentication at all). You always send the auth()-command to the database you're connected to. If it is possible to have the user crendentials stored somewhere special, this needs to be configured on the mongoD side, not on drivers side.

I really hope I did get that right.. 😉

**Comment by FroznMine** [comment]
You can indeed authenticate to a different database (at least in MongoDB 3+). See this page from the official documentation: https://docs.mongodb.com/manual/core/security-users/#user-authentication-database
Here https://docs.mongodb.com/manual/reference/program/mongo/#mongo-shell-authentication-options it's said that the user will be authenticated at the given database when no authentication database is specified.

So basically: yes, you can specify the authentication database separately. And as I said I would really appreciate it.

**Comment by sboesebeck** [comment]
ah... now I get you...

you are right, this is not possible with morphium for the moment. I will add it to the list of features for the next release...

cheers,

Stephan

**Comment by sboesebeck** [comment]
The authentication database cannot be specified to the current driver - at least I could not find something like that now - do you know, how the official mongodb driver is building this kind of behavior?

**Comment by sboesebeck** [comment]
it could also be, that this is done by the mongoD, not the client
